components/sup/src/main.rs

```diff
@@ -222,33 +218,32 @@ fn mgrcfg_from_matches(m: &ArgMatches) -> Result<ManagerConfig> {
         gossip_permanent: m.is_present("PERMANENT_PEER"),
         ring_key: get_ring_key(m)?,
         gossip_peers: get_peers(m)?,
-        gossip_listen: GossipListenAddr::from_str(
-            m.value_of("LISTEN_GOSSIP")
-                .expect("LISTEN_GOSSIP should always have a value."),
-        )?,
-        http_listen: http_gateway::ListenAddr::from_str(
-            m.value_of("LISTEN_HTTP")
-                .expect("LISTEN_HTTP should always have a value."),
-        )?,
-        ctl_listen: ListenCtlAddr::from_str(
-            m.value_of("LISTEN_CTL")
-                .expect("LISTEN_CTL should always have a value."),
-        )?,
+        gossip_listen: m
+            .value_of("LISTEN_GOSSIP")
+            .expect("LISTEN_GOSSIP should always have a value.")
+            .parse()?,
+        http_listen: m
+            .value_of("LISTEN_HTTP")
+            .expect("LISTEN_HTTP should always have a value.")
+            .parse()?,
+        ctl_listen: m
+            .value_of("LISTEN_CTL")
+            .expect("LISTEN_CTL should always have a value.")
+            .parse()?,
+        watch_peer_file: m.value_of("PEER_WATCH_FILE").map(|file| file.to_string()),
+        eventsrv_group: m.value_of("EVENTS").and_then(|events| events.parse().ok()),
+        tls_files: m.value_of("KEY_FILE").and_then(|kf| {
+            Some((
+                PathBuf::from(kf),
+                PathBuf::from(
+                    m.value_of("CERT_FILE")
+                        .expect("CERT_FILE should always have a value if KEY_FILE has a value."),
+                ),
+            ))
+        }),
```

Comment:
As an aside, I think the only thing we still need the `..Default::default()` bit for is for `custom_state_path`, which is something I'd eventually like to remove... it appears to only exist for manipulating testing data.

Might be good to add a note to that effect here.